### ISSUE TITLE
A warning message is output in the new version of Xcode.

### ISSUE THREAD

**Issue by HyunjoonKo** [facebook/facebook-ios-sdk#1295]
## Checklist

- [✓] I've updated to the latest released version of the SDK
- [✓] I've searched for existing [GitHub issues](https://github.com/facebook/facebook-ios-sdk/issues)
- [✓] I've looked for existing answers on [Stack Overflow](https://facebook.stackoverflow.com), the [Facebook Developer Community Forum](https://developers.facebook.com/community/) and the [Facebook Developers Group](https://www.facebook.com/groups/fbdevelopers)
- [✓] I've read the [Code of Conduct](CODE_OF_CONDUCT.md)
- [✓] This issue is not security related and can safely be disclosed publicly on GitHub

## Environment

Describe your dev environment here, giving as many details as possible. If you have them, make sure to include:

- Xcode Version: `11.4.1`
- Swift Version: `5.2.2` (if issue is Swift related)
- Installation Platform & Verison:
-- Carthage Install: `Facebook SDK 6.5.0`
-- Manual Install: `Audience Network SDK 5.8.0`

## Goals

FBAudienceNetwork warning in new version of Xcode.
I want to remove this warning.

## Expected Results

FBAudienceNetwork.framework outputs a large amount of warnings when building in Xcode.

## Actual Results

There is no stack trace because it is a warning at build time.

## Steps to Reproduce

I use the Audience Network as one of AdMob's advertising mediations.
AdMob uses cocoapods, facebook SDK is carthage, FacebookAdapter and FBAudienceNetwork are static libraries.
If I build in Xcode after setting as above, warnings that were not seen in previous versions of Xcode are displayed at build time.

## Code Samples & Details

Build time messages:
```
Showing Recent Messages
Ignoring invalid debug info in /Users/{...}/AdMob Adaptors/FBAudienceNetwork.framework/FBAudienceNetwork(FBAdKVOController.m.o)

Ignoring invalid debug info in /Users/{...}/AdMob Adaptors/FBAudienceNetwork.framework/FBAudienceNetwork(FNFDataBlockDistributor.mm.o)

Ignoring invalid debug info in /Users/{...}/AdMob Adaptors/FBAudienceNetwork.framework/FBAudienceNetwork(FNFMp4AtomManager.mm.o)

Ignoring invalid debug info in /Users/{...}/AdMob Adaptors/FBAudienceNetwork.framework/FBAudienceNetwork(FNFMp4StreamingParser.mm.o)

Ignoring invalid debug info in /Users/{...}/AdMob Adaptors/FBAudienceNetwork.framework/FBAudienceNetwork(FNFVodMp4FrameMetadataCompressor.mm.o)

Ignoring invalid debug info in /Users/{...}/AdMob Adaptors/FBAudienceNetwork.framework/FBAudienceNetwork(FNFDashTrackEntryContainer.mm.o)

Ignoring invalid debug info in /Users/{...}/AdMob Adaptors/FBAudienceNetwork.framework/FBAudienceNetwork(FNFMediaPresentationDescription.mm.o)

Ignoring invalid debug info in /Users/{...}/AdMob Adaptors/FBAudienceNetwork.framework/FBAudienceNetwork(FNFDecompressionHelpersForAVC1.mm.o)

Ignoring invalid debug info in /Users/{...}/AdMob Adaptors/FBAudienceNetwork.framework/FBAudienceNetwork(FNFDecompressionSessionDebuggerForAVC1.mm.o)

Ignoring invalid debug info in /Users/{...}/AdMob Adaptors/FBAudienceNetwork.framework/FBAudienceNetwork(FBBDHelpers.m.o)

Ignoring invalid debug info in /Users/{...}/AdMob Adaptors/FBAudienceNetwork.framework/FBAudienceNetwork(FBBDSignalCollectionExecutorBuilder.m.o)

Ignoring invalid debug info in /Users/{...}/AdMob Adaptors/FBAudienceNetwork.framework/FBAudienceNetwork(unzip.c.o)
```

details:
```
!dbg attachment points at wrong subprogram for function
!72 = distinct !DISubprogram(name: "-[_FBAdKVOInfo debugDescription]", scope: !12, file: !12, line: 190, type: !16, scopeLine: 190, flags: DIFlagPrototyped, spFlags: DISPFlagLocalToUnit | DISPFlagDefinition | DISPFlagOptimized, unit: !11, retainedNodes: !13)
%0* (%2*, i8*)* @"\01-[_FBAdKVOInfo debugDescription]"
  br i1 %47, label %48, label %28, !dbg !86, !llvm.loop !105
!106 = !DILocation(line: 83, column: 5, scope: !81)
!81 = distinct !DISubprogram(name: "describe_options", scope: !12, file: !12, line: 79, type: !16, scopeLine: 80, flags: DIFlagPrototyped, spFlags: DISPFlagLocalToUnit | DISPFlagDefinition | DISPFlagOptimized, unit: !11, retainedNodes: !13)
!81 = distinct !DISubprogram(name: "describe_options", scope: !12, file: !12, line: 79, type: !16, scopeLine: 80, flags: DIFlagPrototyped, spFlags: DISPFlagLocalToUnit | DISPFlagDefinition | DISPFlagOptimized, unit: !11, retainedNodes: !13)
ld: warning: ignoring invalid debug info in /Users/{...}/AdMob Adaptors/FBAudienceNetwork.framework/FBAudienceNetwork(FBAdKVOController.m.o)

!dbg attachment points at wrong subprogram for function
!121 = distinct !DISubprogram(name: "-[FNFDataBlockDistributor moveHeadTo:withHeadType:]", scope: !12, file: !12, line: 101, type: !16, scopeLine: 101, flags: DIFlagPrototyped, spFlags: DISPFlagLocalToUnit | DISPFlagDefinition | DISPFlagOptimized, unit: !11, retainedNodes: !13)
void (%1*, i8*, i64, i64)* @"\01-[FNFDataBlockDistributor moveHeadTo:withHeadType:]"
  br i1 %76, label %77, label %72, !dbg !220, !llvm.loop !221
!222 = !DILocation(line: 141, column: 5, scope: !217)
!217 = distinct !DISubprogram(name: "__tree_min<std::__1::__tree_node_base<void *> *>", scope: !50, file: !50, line: 139, type: !16, scopeLine: 140, flags: DIFlagPrototyped, spFlags: DISPFlagLocalToUnit | DISPFlagDefinition | DISPFlagOptimized, unit: !11, retainedNodes: !13)
!217 = distinct !DISubprogram(name: "__tree_min<std::__1::__tree_node_base<void *> *>", scope: !50, file: !50, line: 139, type: !16, scopeLine: 140, flags: DIFlagPrototyped, spFlags: DISPFlagLocalToUnit | DISPFlagDefinition | DISPFlagOptimized, unit: !11, retainedNodes: !13)
ld: warning: ignoring invalid debug info in /Users/{...}/AdMob Adaptors/FBAudienceNetwork.framework/FBAudienceNetwork(FNFDataBlockDistributor.mm.o)

!dbg attachment points at wrong subprogram for function
!1031 = distinct !DISubprogram(name: "vector", scope: !78, file: !78, line: 1243, type: !16, scopeLine: 1245, flags: DIFlagPrototyped, spFlags: DISPFlagDefinition | DISPFlagOptimized, unit: !11, retainedNodes: !13)
%"class.std::__1::vector.45"* (%"class.std::__1::vector.45"*, %"class.std::__1::vector.45"*)* @_ZNSt3__16vectorINS0_IhNS_9allocatorIhEEEENS1_IS3_EEEC2ERKS5_
  br i1 %27, label %28, label %20, !dbg !1051, !llvm.loop !1064
!1065 = !DILocation(line: 1645, column: 13, scope: !1049)
!1049 = distinct !DISubprogram(name: "__construct_range_forward<std::__1::vector<unsigned char, std::__1::allocator<unsigned char> > *, std::__1::vector<unsigned char, std::__1::allocator<unsigned char> > *>", scope: !593, file: !593, line: 1643, type: !16, scopeLine: 1644, flags: DIFlagPrototyped, spFlags: DISPFlagLocalToUnit | DISPFlagDefinition | DISPFlagOptimized, unit: !11, retainedNodes: !13)
!1049 = distinct !DISubprogram(name: "__construct_range_forward<std::__1::vector<unsigned char, std::__1::allocator<unsigned char> > *, std::__1::vector<unsigned char, std::__1::allocator<unsigned char> > *>", scope: !593, file: !593, line: 1643, type: !16, scopeLine: 1644, flags: DIFlagPrototyped, spFlags: DISPFlagLocalToUnit | DISPFlagDefinition | DISPFlagOptimized, unit: !11, retainedNodes: !13)
ld: warning: ignoring invalid debug info in /Users/{...}/AdMob Adaptors/FBAudienceNetwork.framework/FBAudienceNetwork(FNFMp4AtomManager.mm.o)

!dbg attachment points at wrong subprogram for function
!2525 = distinct !DISubprogram(name: "FNFHasSamples", scope: !12, file: !12, line: 1130, type: !17, scopeLine: 1131, flags: DIFlagPrototyped, spFlags: DISPFlagDefinition | DISPFlagOptimized, unit: !11, retainedNodes: !13)
i1 (%struct.FNFMP4MoofAtom.4109*)* @FNFHasSamples
  br i1 %15, label %22, label %16, !dbg !2570, !llvm.loop !2571
!2572 = !DILocation(line: 832, column: 5, scope: !2568)
!2568 = distinct !DISubprogram(name: "any_of<std::__1::__wrap_iter<const FNFMP4TrunAtom *>, (lambda at Libraries/FNFMp4Parser/FNFMp4Parser/FNFMp4StreamingParser.mm:1137:43)>", scope: !2547, file: !2547, line: 830, type: !17, scopeLine: 831, flags: DIFlagPrototyped, spFlags: DISPFlagLocalToUnit | DISPFlagDefinition | DISPFlagOptimized, unit: !11, retainedNodes: !13)
!2568 = distinct !DISubprogram(name: "any_of<std::__1::__wrap_iter<const FNFMP4TrunAtom *>, (lambda at Libraries/FNFMp4Parser/FNFMp4Parser/FNFMp4StreamingParser.mm:1137:43)>", scope: !2547, file: !2547, line: 830, type: !17, scopeLine: 831, flags: DIFlagPrototyped, spFlags: DISPFlagLocalToUnit | DISPFlagDefinition | DISPFlagOptimized, unit: !11, retainedNodes: !13)
ld: warning: ignoring invalid debug info in /Users/{...}/AdMob Adaptors/FBAudienceNetwork.framework/FBAudienceNetwork(FNFMp4StreamingParser.mm.o)

!dbg attachment points at wrong subprogram for function
!429 = distinct !DISubprogram(name: "seek", scope: !49, file: !49, line: 307, type: !53, scopeLine: 308, flags: DIFlagPrototyped, spFlags: DISPFlagDefinition | DISPFlagOptimized, unit: !48, retainedNodes: !50)
i1 (%class.FNFVodMp4FrameMetadataCompressor*, i32, %struct.FNFMP4FrameMetadata.4126*, i64*)* @_ZNK32FNFVodMp4FrameMetadataCompressor4seekEjR19FNFMP4FrameMetadataRm
  br i1 %34, label %35, label %21, !dbg !465, !llvm.loop !480
!481 = !DILocation(line: 4097, column: 5, scope: !459)
!459 = distinct !DISubprogram(name: "__lower_bound<(lambda at Libraries/FNFMp4Parser/FNFMp4Parser/FNFVodMp4FrameMetadataCompressor.mm:315:30) &, std::__1::__wrap_iter<const FNFMP4Keyframe *>, unsigned int>", scope: !460, file: !460, line: 4093, type: !53, scopeLine: 4094, flags: DIFlagPrototyped, spFlags: DISPFlagLocalToUnit | DISPFlagDefinition | DISPFlagOptimized, unit: !48, retainedNodes: !50)
!459 = distinct !DISubprogram(name: "__lower_bound<(lambda at Libraries/FNFMp4Parser/FNFMp4Parser/FNFVodMp4FrameMetadataCompressor.mm:315:30) &, std::__1::__wrap_iter<const FNFMP4Keyframe *>, unsigned int>", scope: !460, file: !460, line: 4093, type: !53, scopeLine: 4094, flags: DIFlagPrototyped, spFlags: DISPFlagLocalToUnit | DISPFlagDefinition | DISPFlagOptimized, unit: !48, retainedNodes: !50)
ld: warning: ignoring invalid debug info in /Users/{...}/AdMob Adaptors/FBAudienceNetwork.framework/FBAudienceNetwork(FNFVodMp4FrameMetadataCompressor.mm.o)

!dbg attachment points at wrong subprogram for function
!162 = distinct !DISubprogram(name: "-[FNFDashTrackEntryContainer sequenceIndexForRelativeTime:]", scope: !12, file: !12, line: 91, type: !16, scopeLine: 91, flags: DIFlagPrototyped, spFlags: DISPFlagLocalToUnit | DISPFlagDefinition | DISPFlagOptimized, unit: !11, retainedNodes: !13)
i32 (%0*, i8*, i64)* @"\01-[FNFDashTrackEntryContainer sequenceIndexForRelativeTime:]"
  br i1 %35, label %36, label %23, !dbg !190, !llvm.loop !206
!207 = !DILocation(line: 4139, column: 5, scope: !182)
!182 = distinct !DISubprogram(name: "__upper_bound<std::__1::__less<unsigned long long, unsigned long long> &, std::__1::__wrap_iter<unsigned long long *>, unsigned long long>", scope: !183, file: !183, line: 4135, type: !16, scopeLine: 4136, flags: DIFlagPrototyped, spFlags: DISPFlagDefinition | DISPFlagOptimized, unit: !11, retainedNodes: !13)
!182 = distinct !DISubprogram(name: "__upper_bound<std::__1::__less<unsigned long long, unsigned long long> &, std::__1::__wrap_iter<unsigned long long *>, unsigned long long>", scope: !183, file: !183, line: 4135, type: !16, scopeLine: 4136, flags: DIFlagPrototyped, spFlags: DISPFlagDefinition | DISPFlagOptimized, unit: !11, retainedNodes: !13)
ld: warning: ignoring invalid debug info in /Users/{...}/AdMob Adaptors/FBAudienceNetwork.framework/FBAudienceNetwork(FNFDashTrackEntryContainer.mm.o)

!dbg attachment points at wrong subprogram for function
!126 = distinct !DISubprogram(name: "+[FNFMediaPresentationDescription presentationDescriptionWithContentsOfXml:mpdUrl:currentTime:error:]", scope: !60, file: !60, line: 327, type: !64, scopeLine: 327, flags: DIFlagPrototyped, spFlags: DISPFlagLocalToUnit | DISPFlagDefinition | DISPFlagOptimized, unit: !59, retainedNodes: !61)
i8* (i8*, i8*, %5*, %6*, %4*, %7**)* @"\01+[FNFMediaPresentationDescription presentationDescriptionWithContentsOfXml:mpdUrl:currentTime:error:]"
  br i1 %1855, label %1856, label %1842, !dbg !2114, !llvm.loop !2125
!2126 = !DILocation(line: 2668, column: 5, scope: !2112)
!2112 = distinct !DISubprogram(name: "__lower_bound<unsigned long long>", scope: !1365, file: !1365, line: 2664, type: !64, scopeLine: 2667, flags: DIFlagPrototyped, spFlags: DISPFlagDefinition | DISPFlagOptimized, unit: !59, retainedNodes: !61)
!2112 = distinct !DISubprogram(name: "__lower_bound<unsigned long long>", scope: !1365, file: !1365, line: 2664, type: !64, scopeLine: 2667, flags: DIFlagPrototyped, spFlags: DISPFlagDefinition | DISPFlagOptimized, unit: !59, retainedNodes: !61)
ld: warning: ignoring invalid debug info in /Users/{...}/AdMob Adaptors/FBAudienceNetwork.framework/FBAudienceNetwork(FNFMediaPresentationDescription.mm.o)

!dbg attachment points at wrong subprogram for function
!210 = distinct !DISubprogram(name: "FNFDecompressionAVC1TrimStartCode", scope: !12, file: !12, line: 240, type: !16, scopeLine: 242, flags: DIFlagPrototyped, spFlags: DISPFlagDefinition | DISPFlagOptimized, unit: !11, retainedNodes: !13)
void (%struct.opaqueCMSampleBuffer.4317*, %struct.opaqueCMSampleBuffer.4317**)* @FNFDecompressionAVC1TrimStartCode
  br i1 %220, label %207, label %210, !dbg !367, !llvm.loop !375
!376 = !DILocation(line: 216, column: 3, scope: !355)
!355 = distinct !DISubprogram(name: "repacketsizeFrame", scope: !12, file: !12, line: 209, type: !16, scopeLine: 213, flags: DIFlagPrototyped, spFlags: DISPFlagLocalToUnit | DISPFlagDefinition | DISPFlagOptimized, unit: !11, retainedNodes: !13)
!355 = distinct !DISubprogram(name: "repacketsizeFrame", scope: !12, file: !12, line: 209, type: !16, scopeLine: 213, flags: DIFlagPrototyped, spFlags: DISPFlagLocalToUnit | DISPFlagDefinition | DISPFlagOptimized, unit: !11, retainedNodes: !13)
ld: warning: ignoring invalid debug info in /Users/{...}/AdMob Adaptors/FBAudienceNetwork.framework/FBAudienceNetwork(FNFDecompressionHelpersForAVC1.mm.o)

!dbg attachment points at wrong subprogram for function
!34 = distinct !DISubprogram(name: "-[FNFDecompressionSessionDebuggerForAVC1 dealloc]", scope: !12, file: !12, line: 36, type: !16, scopeLine: 36, flags: DIFlagPrototyped, spFlags: DISPFlagLocalToUnit | DISPFlagDefinition | DISPFlagOptimized, unit: !11, retainedNodes: !13)
void (%0*, i8*)* @"\01-[FNFDecompressionSessionDebuggerForAVC1 dealloc]"
  br i1 %44, label %45, label %40, !dbg !115, !llvm.loop !116
!117 = !DILocation(line: 141, column: 5, scope: !112)
!112 = distinct !DISubprogram(name: "__tree_min<std::__1::__tree_node_base<void *> *>", scope: !38, file: !38, line: 139, type: !16, scopeLine: 140, flags: DIFlagPrototyped, spFlags: DISPFlagLocalToUnit | DISPFlagDefinition | DISPFlagOptimized, unit: !11, retainedNodes: !13)
!112 = distinct !DISubprogram(name: "__tree_min<std::__1::__tree_node_base<void *> *>", scope: !38, file: !38, line: 139, type: !16, scopeLine: 140, flags: DIFlagPrototyped, spFlags: DISPFlagLocalToUnit | DISPFlagDefinition | DISPFlagOptimized, unit: !11, retainedNodes: !13)
ld: warning: ignoring invalid debug info in /Users/{...}/AdMob Adaptors/FBAudienceNetwork.framework/FBAudienceNetwork(FNFDecompressionSessionDebuggerForAVC1.mm.o)

!dbg attachment points at wrong subprogram for function
!200 = distinct !DISubprogram(name: "encryptJsonData", scope: !12, file: !12, line: 234, type: !16, scopeLine: 235, flags: DIFlagPrototyped, spFlags: DISPFlagDefinition | DISPFlagOptimized, unit: !11, retainedNodes: !13)
%0* (%6*, %0*)* @encryptJsonData
  br i1 %120, label %109, label %101, !dbg !260, !llvm.loop !274
!275 = !DILocation(line: 344, column: 3, scope: !240)
!240 = distinct !DISubprogram(name: "xor", scope: !12, file: !12, line: 319, type: !16, scopeLine: 320, flags: DIFlagPrototyped, spFlags: DISPFlagLocalToUnit | DISPFlagDefinition | DISPFlagOptimized, unit: !11, retainedNodes: !13)
!240 = distinct !DISubprogram(name: "xor", scope: !12, file: !12, line: 319, type: !16, scopeLine: 320, flags: DIFlagPrototyped, spFlags: DISPFlagLocalToUnit | DISPFlagDefinition | DISPFlagOptimized, unit: !11, retainedNodes: !13)
ld: warning: ignoring invalid debug info in /Users/{...}/AdMob Adaptors/FBAudienceNetwork.framework/FBAudienceNetwork(FBBDHelpers.m.o)

!dbg attachment points at wrong subprogram for function
!2291 = distinct !DISubprogram(name: "__isJailBrokenInternalBlock_block_invoke", scope: !12, file: !12, line: 1558, type: !17, scopeLine: 1558, flags: DIFlagPrototyped, spFlags: DISPFlagLocalToUnit | DISPFlagDefinition | DISPFlagOptimized, unit: !11, retainedNodes: !13)
void (i8*, void (%7*)*)* @__isJailBrokenInternalBlock_block_invoke
  br i1 %114, label %124, label %115, !dbg !2369, !llvm.loop !2370
!2371 = !DILocation(line: 1826, column: 3, scope: !2365)
!2365 = distinct !DISubprogram(name: "hasLibrariesInjected", scope: !12, file: !12, line: 1822, type: !17, scopeLine: 1822, spFlags: DISPFlagLocalToUnit | DISPFlagDefinition | DISPFlagOptimized, unit: !11, retainedNodes: !13)
!2365 = distinct !DISubprogram(name: "hasLibrariesInjected", scope: !12, file: !12, line: 1822, type: !17, scopeLine: 1822, spFlags: DISPFlagLocalToUnit | DISPFlagDefinition | DISPFlagOptimized, unit: !11, retainedNodes: !13)
ld: warning: ignoring invalid debug info in /Users/{...}/AdMob Adaptors/FBAudienceNetwork.framework/FBAudienceNetwork(FBBDSignalCollectionExecutorBuilder.m.o)

!dbg attachment points at wrong subprogram for function
!21 = distinct !DISubprogram(name: "unzOpenInternal", scope: !6, file: !6, line: 388, type: !10, scopeLine: 389, flags: DIFlagPrototyped, spFlags: DISPFlagLocalToUnit | DISPFlagDefinition | DISPFlagOptimized, unit: !5, retainedNodes: !7)
i8* (i8*, %struct.fb_zlib_filefunc64_32_def_s.5281*)* @unzOpenInternal
  br i1 %86, label %65, label %87, !dbg !87, !llvm.loop !108
!109 = !DILocation(line: 333, column: 9, scope: !61)
!61 = distinct !DISubprogram(name: "unz64local_SearchCentralDir", scope: !6, file: !6, line: 293, type: !10, scopeLine: 294, flags: DIFlagPrototyped, spFlags: DISPFlagLocalToUnit | DISPFlagDefinition | DISPFlagOptimized, unit: !5, retainedNodes: !7)
!61 = distinct !DISubprogram(name: "unz64local_SearchCentralDir", scope: !6, file: !6, line: 293, type: !10, scopeLine: 294, flags: DIFlagPrototyped, spFlags: DISPFlagLocalToUnit | DISPFlagDefinition | DISPFlagOptimized, unit: !5, retainedNodes: !7)
ld: warning: ignoring invalid debug info in /Users/{...}/AdMob Adaptors/FBAudienceNetwork.framework/FBAudienceNetwork(unzip.c.o)
```

**Comment by hovox** [comment]
Hi ,
We are trying to update to Xcode 11.4.1 and got those warnings. Is there any way to fix them?
@HyunjoonKo have you found the way how to fix the issue?

**Comment by HyunjoonKo** [comment]
@hovox I updated to Facebook SDK 6.5.0 and Xcode 11.4.1, but the warning did not disappear.

**Comment by 0x0c** [comment]
I got same warnings and can't find any solutions.

**Comment by HermanKayy** [comment]
Any solutions?

**Comment by ManueGE** [comment]
Any solutions?

**Comment by tinder-owenthomas** [comment]
Any progress on this one? We're held up on adopting Xcode 11.4 until this is fixed.

**Comment by HyunjoonKo** [comment]
There is no workaround for the warnings at build time.
The SDK has been updated, but there are still warnings:
 - Xcode 11.5
 - Facebook SDK 7.0.1
 - Facebook audience network 5.9.0
 - AdMob/FacebookAdapter 5.9.0.1

**Comment by ironyk** [comment]
The issue appears to be related to Xcode 11.4+. It occurs for every AudienceNetwork SDK built with Xcode 11.3 or earlier (which is every released version at this moment).
There is a similar issue raised on Apple's developer forum with more details: https://forums.developer.apple.com/thread/130677

The next SDK release will resolve the issue. It is planned for the beginning of June

**Comment by gui17aume** [comment]
Hi @ironyk,
In the meantime, couldn't you release a corrective version of FBAudienceNetwork (let's say 5.9.1) by building the exact same code as 5.9.0 but using Xcode 11.4+?
It would be a relief for a lot of developers 😌

**Comment by ironyk** [comment]
HI @gui17aume, 
If I read the issue correctly, it is about debug info and debugger being unable to work properly with certain classes within FBAudienceNetwork. I believe it does not produce any side effects.

Could you please provide more details on how and why this affects your release / development?

If the only reason to release a patch is to silence the warnings, I'm afraid we cannot prioritize this at the moment

**Comment by gui17aume** [comment]
Hi @ironyk,
Yes the warnings are the main issue.

When you work on a project containing thousands of source files and you make efforts everyday to keep the compilation warning-free, it is very annoying that an up-to-date external library provided by one of the biggest tech companies in the world triggers no less than 13 warnings for almost 2 months now 😕 

Among the dozens of SDK we're using, I'm afraid FBAudienceNetwork is the only one having this issue.

**Comment by HyunjoonKo** [comment]
Now the warning problem does not appear. It seems to be resolved.

I am using the following environment:
- Xcode 11.6
- Facebook SDK 7.1.1
- Facebook audience Network SDK 5.10.1
- AdMob/FacebookAdapter 5.10.1.0

**Comment by tinder-owenthomas** [comment]
@HyunjoonKo I tried updating to FB Audience Network 5.10.1 and Xcode now complains that it has internal deprecations (they're both deprecating and simultaneously using `FBInstreamAdView`). Have you encountered this?

Warning: `'FBInstreamAdView' is deprecated: Instream ads have been deprecated.`

**Comment by HyunjoonKo** [comment]
@tinder-owenthomas I am currently using SDK 6.0.0, so I didn't see any warning messages.

**Comment by tinder-owenthomas** [comment]
Ah, ok. I'm not sure if our business is ready to update to a new major version before the holidays. I'll check with them though. Thank you!

**Comment by tinder-owenthomas** [comment]
@HyunjoonKo I'm trying to upgrade to 6.2.1 to be able to support iOS 14 App Tracking Transparency and I still cannot get it to work because I'm getting the same deprecation errors. They're actually Xcode errors so they keep it from even compiling!

Are you having any issues going to 6.2.1? Seems this version is necessary to support ATT.

**Comment by HyunjoonKo** [comment]
Hi @tinder-owenthomas 
I'm currently using SDK version 8.2.0, so I haven't seen that error.

**Comment by tinder-owenthomas** [comment]
Oh, are you talking about the main Facebook SDK? Or FBAudienceNetwork specifically? FBAudienceNetwork is the one that has the problems and is only up to 6.2.1 at this time.

**Comment by HyunjoonKo** [comment]
@tinder-owenthomas 

Oh, sorry. I made a mistake.
I am using Audience Network version 6.2.1 and a warning message is displayed.

`FBInstreamAdView` has been deprecated since version 6.0.0, but I don't care because I don't use instream.
I think it will disappear when `FBInstreamAdView` is completely removed.

**Comment by tinder-owenthomas** [comment]
The problem is our project will not archive if there are compile errors like this, so it is keeping us from releasing an updated app that will support iOS ATT sounds like it will be fixed in 6.3.0 so hopefully that comes before the ATT deadline 🤞
